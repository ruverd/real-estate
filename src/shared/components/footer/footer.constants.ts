export const MAP_TEXTS = {
  developedBy: "Developed by Ruver Dornelas",
};
