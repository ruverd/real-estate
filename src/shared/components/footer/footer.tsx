import { MAP_TEXTS } from "./footer.constants";

export const Footer = () => {
  return (
    <footer className="py-4 mt-auto border-t bg-background text-foreground text-sm">
      <div className="container mx-auto flex items-center justify-center gap-2">
        <span>{MAP_TEXTS.developedBy}</span>
      </div>
    </footer>
  );
};
