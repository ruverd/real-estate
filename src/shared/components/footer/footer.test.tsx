import { render, screen } from '@testing-library/react';

import { Footer } from './footer';
import { MAP_TEXTS } from './footer.constants';

const renderFooter = () => {
  return render(<Footer />)
}

describe('Footer Component', () => {
  it('should renders correctly', () => {
    const { container } = renderFooter();
    
    const footerElement = container.querySelector('footer');
    expect(footerElement).toBeInTheDocument();
    
    expect(screen.getByText(MAP_TEXTS.developedBy)).toBeInTheDocument();
  });
  
  it('should has the correct styling classes', () => {
    const { container } = renderFooter();

    const footerElement = container.querySelector('footer');
    
    expect(footerElement).toHaveClass('bg-background');
    expect(footerElement).toHaveClass('text-foreground');
  });
  
  it('should match snapshot', () => {
    const { container } = renderFooter();

    expect(container.firstChild).toMatchSnapshot();
  });
});
