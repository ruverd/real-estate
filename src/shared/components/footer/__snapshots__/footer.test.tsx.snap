// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > should match snapshot 1`] = `
<footer
  class="py-4 mt-auto border-t bg-background text-foreground text-sm"
>
  <div
    class="container mx-auto flex items-center justify-center gap-2"
  >
    <span>
      Developed by Ruver Dornelas
    </span>
  </div>
</footer>
`;
